### ISSUE TITLE
Training with only a few but high resolution image pairs

### ISSUE THREAD

**Issue by yingzwang** [junyanz/pytorch-CycleGAN-and-pix2pix#416]
I want to generate watercolor paintings from pencil sketches. I have perfectly paired training data, but only a few pairs. 

Q: When we have paired training data, pix2pix always works better than CycleGAN. Is that correct?
  
I trained pix2pix model on only two training pairs, and the output is amazing. Please see below my training pairs, and the generated new paintings.
![training-pair1](https://user-images.githubusercontent.com/38169187/47744047-a6f81180-dc80-11e8-992d-5878afb479dd.png)
![training-pari2](https://user-images.githubusercontent.com/38169187/47744090-c131ef80-dc80-11e8-9e76-38ddca7ac100.png)
![output_paintings](https://user-images.githubusercontent.com/38169187/47744226-1e2da580-dc81-11e8-8540-e1caaa3b39fb.png)
The results above are obtained by training on images of size 512 x 512. The original scans of my paintings have much higher resolution, i.e., 4096 x 4096. I feel that the results can get much better if I take advantage of the high resolution images. 

Q: Will it help if I apply a random crop on the original image of size 4096 x 4096, to create many 512 x 512 training pairs?

**Comment by junyanz** [comment]
Your results look really cool. Data augmentation might help! See this paper for training a pix2pix [model](https://arxiv.org/abs/1805.04487) with only one image. They have some tricks!

**Comment by yingzwang** [comment]
Thanks for your reply. I'll check the paper and try their tricks.

Another question. The GAN is supposed to learn a distribution conditioned on the input, but my output seems quite deterministic. Is it due to my limited training data, or the limitation of pix2pix? How can I generate more random and different outputs from the same input?

**Comment by yingzwang** [comment]
Thanks for your reply. I'll check the paper and try their tricks.
